### PR TITLE
 🌱 Cleanup owner files

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
As reminded in this week's SIG chair and lead meeting, it is important to keep our owner files up to date.

I made a pass based on [reviewer stats](https://k8s.devstats.cncf.io/d/13/developer-activity-counts-by-repository-group?orgId=1&var-period_name=Last+year&var-metric=reviews&var-repogroup_name=All&var-repo_name=kubernetes-sigs%2Fcluster-api&var-country_name=All&from=1558110159436&to=1715962959436&viewPanel=5). 

If by chance I made some error, please don't hesitate to reach out.

Let me take this moment to thank you again emeritus reviewers for all the great work in the project so far:
- @stmcginnis 
- @dongsupark 
- @invidian
- @johananl
- @oscr 

/area security
/hold let's give folks time to comment on this PR + I will bring it up next week in the office hour

### CODE REVIEW

**Comment by johananl** [comment]
Thanks @fabriziopandini. I am still involved with CAPI but at the moment am only working on a design proposal in a Google document so not much GitHub activity lately. Personally I'm fine with being removed from the owners file. We can always reevaluate in the future if necessary.

**Comment by stmcginnis** [comment]
Sadly it's been awhile since I've been able to spend much time on CAPI. Maybe that will change one day, but until then:

/lgtm

**Review by chrischdi — COMMENTED** [review]
/lgtm

Thanks to everyone for the past contributions and happy to welcome anyone back for future contributions :-)

**Review by vincepri — APPROVED** [review]

**Comment by sbueringer** [comment]
@fabriziopandini Once we merge this PR, can we please sync back the OWNER file? Just saw that previous approvers are stil getting assigned on PRs on 1.7 + 1.6 release branches

**Comment by fabriziopandini** [comment]
Lazy consensus expired
/hold cancel

**Comment by sbueringer** [comment]
Probably will need manual cherry-picks. I think the files are already out of sync

**Comment by sbueringer** [comment]
Nevermind, one cherry-pick worked :D

**Pull request merged**
